src/ceph-volume/ceph_volume/systemd/main.py

```diff
@@ -0,0 +1,91 @@
+"""
+This file is used only by systemd units that are passing their instance suffix
+as arguments to this script so that it can parse the suffix into arguments that
+``ceph-volume <sub command>`` can consume
+"""
+import os
+import sys
+import time
+import logging
+from ceph_volume import log, process
+from ceph_volume.exceptions import SuffixParsingError
+
+
+def parse_subcommand(string):
+    subcommand = string.rsplit('-', 1)[-1]
+    if not subcommand:
+        raise SuffixParsingError('subcommand', string)
+    return subcommand
+
+
+def parse_osd_id(string):
+    osd_id = string.split('-', 1)[0]
+    if not osd_id:
+        raise SuffixParsingError('OSD id', string)
+    if osd_id.isdigit():
+        return osd_id
+    raise SuffixParsingError('OSD id', string)
+
+
+def parse_osd_uuid(string):
+    osd_id = '%s-' % parse_osd_id(string)
+    osd_subcommand = '-%s' % parse_subcommand(string)
+    # remove the id first
+    trimmed_suffix = string.split(osd_id)[-1]
+    # now remove the sub command
+    osd_uuid = trimmed_suffix.split(osd_subcommand)[0]
+    if not osd_uuid:
+        raise SuffixParsingError('OSD uuid', string)
+    return osd_uuid
+
+
+def main(args=None):
+    """
+    Main entry point for the ``ceph-volume-systemd`` executable. ``args`` are
+    optional for easier testing of arguments.
+
+    Expected input is similar to::
+
+        ['/path/to/ceph-volume-systemd', '<osd id>-<osd uuid>-<device type>']
```

Comment:
Ok, I see where you are going. I agree here that it is useful to allow for extra/non-strict stuff. Since I was relying on correct argument passing to `ceph-volume lvm activate` I think it might be reasonable to pass a flag that indicates a "blob" is getting passed instead. Maybe something like:

    ceph-volume@lvm-sha1-osd_id

Would end up calling:

    ceph-volume lvm activate --systemd sha1-osd_id

In a similar way then, nvme could be:

    ceph-volume@nvme-foo-100-bar-1

Would go then to:

    ceph-volume nvme activate --systemd foo-100-bar-1

Does that sound flexible enough?